homeassistant/components/luxtronik/config_flow.py

```diff
@@ -0,0 +1,141 @@
+"""Config flow to configure the Luxtronik heatpump controller integration."""
+# region Imports
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.dhcp import HOSTNAME, IP_ADDRESS, DhcpServiceInfo
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_LANGUAGE_SENSOR_NAMES,
+    CONF_LOCK_TIMEOUT,
+    CONF_SAFE,
+    CONF_UPDATE_IMMEDIATELY_AFTER_WRITE,
+    DEFAULT_PORT,
+    DOMAIN,
+    LANG_DEFAULT,
+    LANGUAGES_SENSOR_NAMES,
+    LOGGER,
+)
+from .helpers.lux_helper import discover
+
+# endregion Imports
+
+
+class LuxtronikFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Luxtronik heatpump controller config flow."""
+
+    VERSION = 1
+    _hassio_discovery = None
+    _discovery_host = None
+    _discovery_port = None
+    _discovery_schema = None
+
+    def _get_schema(self):
+        return vol.Schema(
+            {
+                vol.Required(CONF_HOST, default=self._discovery_host): str,
+                vol.Required(CONF_PORT, default=self._discovery_port): int,
+                vol.Optional(CONF_LANGUAGE_SENSOR_NAMES, default=LANG_DEFAULT): vol.In(
+                    LANGUAGES_SENSOR_NAMES
+                ),
```

Comment:
I don't think setting a language is needed here.
We generally bring in names in English and allow the user to customize them at will.